src/vm/node_modules/vmload/vmload-datasets.js

```diff
@@ -261,6 +261,15 @@ function getDatasets(options, callback)
         log: log,
         types: zfs_types
     };
+    /*
+     * If this is about one specific zone, and it's a smartos/lx zone, we only need
+     * to do our zfs list recursively under the zone's dataset.
+     *
+     * This saves a lot of time on machines with lots of zfs datasets and snaps
+     */
+    if (uuid && (brand === 'joyent' || brand === 'joyent-minimal' || brand === 'lx')) {
+        _options.parent = 'zones/' + uuid;
```

Comment:
I believe it is possible to have SmartOS use a different root dataset than "zones"... this may break for those users.